src/Features/CSharp/Portable/ConvertSwitchStatementToExpression/ConvertSwitchStatementToExpressionCodeFixProvider.Rewriter.cs

```diff
@@ -0,0 +1,106 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using Microsoft.CodeAnalysis.CSharp.Extensions;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.CSharp.ConvertSwitchStatementToExpression
+{
+    using static SyntaxFactory;
+
+    internal sealed partial class ConvertSwitchStatementToExpressionCodeFixProvider
+    {
+        private sealed class Rewriter : CSharpSyntaxVisitor<SyntaxNode>
+        {
+            private List<ExpressionSyntax> _assignmentTargetsOpt;
```

Comment:
instead of mutable and opt.  why not just be readonly and always initialized.  on the consumption side, empty would mean 'not assigning'.  just a though.

Reply:
this is to not allocate if it's all `return`. might be not that important.

Reply:
Yeah i would ditch it. :)